src/ui/pages/SendTransaction/SendTransaction.tsx

```diff
@@ -370,166 +587,190 @@ function TransactionDefaultView({
-  const chainId = transaction ? normalizeChainId(transaction.chainId) : null;
-  const { networks, loadNetworkByChainId } = useNetworks();
-  useEffect(() => {
-    if (chainId) {
-      loadNetworkByChainId(chainId);
-    }
-  }, [chainId, loadNetworkByChainId]);
-  const chain = (chainId ? networks?.getChainById(chainId) : null) || null;
+  const [configuration, setConfiguration] = useState(DEFAULT_CONFIGURATION);
+  const { data: chainGasPrices } = useGasPrices(chain);
 
-  const transactionAction =
-    transaction && networks && chain
-      ? describeTransaction(transaction, {
-          networks,
-          chain,
-        })
-      : null;
+  const transactionAction = describeTransaction(populatedTransaction, {
+    networks,
+    chain,
+  });
 
-  const { data: localAddressAction, ...localAddressActionQuery } = useQuery({
-    queryKey: [
-      'incomingTxToIncomingAddressAction',
-      transaction,
+  const { data: localAddressAction, ...localAddressActionQuery } =
+    useLocalAddressAction({
+      address: singleAddress,
       transactionAction,
+      transaction: populatedTransaction,
       networks,
-      singleAddress,
-    ],
-    queryFn: () => {
-      return transaction && networks && transactionAction
-        ? incomingTxToIncomingAddressAction(
-            {
-              transaction: { ...transaction, from: singleAddress },
-              hash: '',
-              timestamp: 0,
-            },
-            transactionAction,
-            networks
-          )
-        : null;
+    });
+
+  const [allowanceQuantityBase, setAllowanceQuantityBase] = useState('');
+
+  const configureTransactionToBeSigned = useEvent(
+    async (transaction: IncomingTransaction) =>
+      configureTransactionToSign(transaction, {
+        chainGasPrices: chainGasPrices || null,
+        configuration,
+        allowanceQuantityBase,
+        chain,
+        from: singleAddress,
+        networks,
+        transactionAction,
+      })
+  );
+
+  const { data: eligibility } = useQuery({
+    suspense: false,
+    staleTime: 120000,
+    queryKey: ['paymaster/check-eligibility', incomingTransaction],
+    queryFn: async () => {
+      const tx = await configureTransactionToBeSigned(incomingTransaction);
+      return checkPaymasterEligibility(tx);
     },
-    keepPreviousData: true,
-    enabled:
-      Boolean(transaction) && Boolean(networks) && Boolean(transactionAction),
-    useErrorBoundary: true,
-    retry: false,
-    refetchOnMount: false,
-    refetchOnReconnect: false,
-    refetchOnWindowFocus: false,
+    enabled: FEATURE_PAYMASTER_ENABLED,
   });
 
-  const { data: interpretation, ...interpretQuery } = useQuery({
+  const paymasterEligible = Boolean(eligibility?.data.eligible);
+
+  const txInterpretQuery = useInterpretTransaction({
+    transaction: populatedTransaction,
+    address: singleAddress,
+    origin,
+    enabled: FEATURE_PAYMASTER_ENABLED
+      ? eligibility?.data.eligible === false
```

Comment:
Do we run interpretation for not eligible transactions?

Reply:
For eligible transactions we interpret typedData instead